### ISSUE TITLE
How can I wait for a subscription to be completed without rebinding it?

### ISSUE THREAD

**Issue by johynpapin** [vuejs/vuefire#850]
Hello,

I need to wait for the completion of one of the vuexfire bindings before calling `next` in the `beforeEach` guard of Vue Router. Like this:

```js
await store.dispatch('subscribeToUserInfo')

if (store.state.userInfo) {
  next()
} else {
  next({ name: 'AuthCompleteSignUp' })
}
```

This method works, but as I understand it, it will unbind the reference each time the `beforeEach` guard is called.

So I have a few questions:

- If I leave it like this, will I have to pay for a read each time the guard is called?
- If I pass `{ reset: false }`, will the current binding be retained?
- If not, am I required to check that the binding does not already exist in the vuex action? Like this:

```js
subscribeToUserInfo: firestoreAction(async ({ state, commit, bindFirestoreRef }) => {
  if (state.userInfo) {
    return
  }

  await bindFirestoreRef('userInfo', firestore.collection('users').doc(state.userId))
})
```

In short, I don't understand what happens if I rebind a reference.

I thank you in advance

**Comment by posva** [comment]
I can't answer Firebase pricing related question, this library is not related to them, and I could answer things wrong.
For other questions, please ask them on the [forum](http://forum.vuejs.org/), the [Discord server](https://vue-land.js.org/) or [StackOverflow](http://stackoverflow.com/tags/vue.js).

`reset` is to keep data while waiting for a binding (https://vuefire.vuejs.org/api/vuefire.html#bind)

When you rebind a reference, it does the same thing as binding a fresh one: listen to changes (get initial value as well).
You can't wait for changes to be finished because it's a stream of data that doesn't end. You should be able to await firebase operations though

**Comment by johynpapin** [comment]
Thank you very much for your prompt response. In that case, I will do as I did in my second code snippet.
